### PR TITLE
RDISCROWD-3797: Encrypt task response

### PR DESCRIPTION
**Pull request by dchhabda**
* encrypt task response under `private_json__encrypted_response` if task payload contains `private_json__encrypted_payload`

### CODE REVIEW

File: pybossa/api/task_run.py [hunk 1]
```diff
@@ -70,14 +72,23 @@ def _preprocess_post_data(self, data):
         project_id = data['project_id']
         self.check_can_post(project_id, task_id)
         preprocess_task_run(project_id, task_id, data)
-        info = data['info']
-        path = "{0}/{1}/{2}".format(project_id, task_id, current_user.id)
         if with_encryption:
```
Comment (kbecker42) [inline pybossa/api/task_run.py:75]: Just wanted to confirm if moving lines 73, 74 into this "if" block was intended, as they will only be executed now if `with_encryption`.

```python
info = data['info']
path = "{0}/{1}/{2}".format(project_id, task_id, current_user.id)
```

File: pybossa/api/task_run.py [hunk 2]
```diff
@@ -70,14 +72,23 @@ def _preprocess_post_data(self, data):
...
+            encrypted_response = encrypt_task_response_data(task_id, project_id, info)
+            data['info'] = {}
+            pyb_answer_url = upload_json_data(json_data=info, upload_path=path,
+                file_name='pyb_answer.json', encryption=with_encryption,
+                conn_name='S3_TASKRUN', upload_root_dir=upload_root_dir
+            )
+            if pyb_answer_url:
+                data['info']['pyb_answer_url'] = pyb_answer_url
+            if encrypted_response:
+                data['info']['private_json_encrypted_response'] = encrypted_response
```
Comment (kbecker42) [inline pybossa/api/task_run.py:90]: Looks good, just confirming the naming of the field should use single underscores or a double.

`private_json_encrypted_response` versus `private_json__encrypted_response`.

**Review by kbecker42 — COMMENTED** [review]
Thank you.

File: pybossa/api/task_run.py (hunk 1 above)
Comment (dchhabda) [reply pybossa/api/task_run.py:75]: info and path generated were used only with_encryption true, hence its safe to set it only when needed. Refer original code [here](https://github.com/bloomberg/pybossa/blob/master/pybossa/api/task_run.py#L73-L80)

File: pybossa/api/task_run.py (hunk 2 above)
Comment (dchhabda) [reply pybossa/api/task_run.py:90]: This is going to be reserved keyword that would be documented. Hence, either single/double underscores can be used.

**Review by kbecker42 — DISMISSED** [review]

**Review by kbecker42 — DISMISSED** [review]

**Review by kbecker42 — APPROVED** [review]

**Pull request merged**